streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java

```diff
@@ -80,12 +81,29 @@ public static void handleDeserializationFailure(final DeserializationExceptionHa
                                                     final ConsumerRecord<byte[], byte[]> rawRecord,
                                                     final Logger log,
                                                     final Sensor droppedRecordsSensor) {
+        handleDeserializationFailure(deserializationExceptionHandler, processorContext, deserializationException, rawRecord, log, droppedRecordsSensor, null);
+    }
+
+    public static void handleDeserializationFailure(final DeserializationExceptionHandler deserializationExceptionHandler,
+                                                    final ProcessorContext<?, ?> processorContext,
+                                                    final Exception deserializationException,
+                                                    final ConsumerRecord<byte[], byte[]> rawRecord,
+                                                    final Logger log,
+                                                    final Sensor droppedRecordsSensor,
+                                                    final String sourceNodeName) {
         final DeserializationExceptionHandler.DeserializationHandlerResponse response;
         try {
-            response = deserializationExceptionHandler.handle(
+            final DefaultErrorHandlerContext errorHandlerContext = new DefaultErrorHandlerContext(
                 (InternalProcessorContext<?, ?>) processorContext,
-                rawRecord,
-                deserializationException);
+                rawRecord.topic(),
+                rawRecord.partition(),
+                rawRecord.offset(),
+                rawRecord.headers(),
+                rawRecord.key(),
+                rawRecord.value(),
+                sourceNodeName,
+                processorContext.taskId());
+            response = deserializationExceptionHandler.handle(errorHandlerContext, rawRecord, deserializationException);
         } catch (final Exception fatalUserException) {
```

Comment:
I realized now that you did not consider that the exception handler can also throw an exception for the processing exception handler (and I as the reviewer also missed it). Could you open a separate PR that catches exceptions from the processing exception handler and re-throws them as it is done here?
Sorry for not noticing it earlier!

Reply:
@cadonna As processing exception handler handles exception at `ProcessorNode`, indeed exceptions during deserialization are not caught right now. Is it your point?

Maybe we should consider catching and handling exceptions when getting the `nextRecord` at `StreamTask` level right before processing it: 

https://github.com/apache/kafka/blob/a5bfc2190c3448039c9361909e547f64f7fdb6e2/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java#L770

Going to sleep on it

Reply:
done in this PR https://github.com/apache/kafka/pull/16675